DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/autorun/src/autorun.hpp

```diff
@@ -0,0 +1,126 @@
+#ifndef __AUTORUN_HPP__
+#define __AUTORUN_HPP__
+
+#include <CL/sycl.hpp>
+#include <type_traits>
+
+/*
+This header defines the Autorun kernel utility. This utility is used to
+launch kernels that are submitted before main begins. It is typically used
+to launch kernels that run forever.
+
+Two classes are defined in this header file: Autorun and AutorunForever.
+Autorun creates an autorun kernel that is NOT implicitly wrapped in an infinite

```

Comment:
I would rephrase this.  AutorunForever creates a kernel and wraps it in a while(1) loop.  Autorun creates the kernel and does not wrap it in a while(1) loop.  Usually when using Autorun the user will have the while(1) loop explicitly in their code.  Something like that.